### PR TITLE
Adds sub tests

### PR DESCRIPTION
**Pull request by dungeonmaster18**
Signed-off-by: dungeonmaster18 <umesh4257@gmail.com>

**What this PR does / why we need it**:
Adds sub test.

- [x] [openebs/openebs#998](https://github.com/openebs/openebs/issues/998)
- [X] [openebs/openebs#1000](https://github.com/openebs/openebs/issues/1000)

Here is the output:

k8s_test.go
```
$ go test -v
=== RUN   TestGetConfigMap
=== RUN   TestGetConfigMap/TestGetConfigMap
--- PASS: TestGetConfigMap (0.00s)
    --- PASS: TestGetConfigMap/TestGetConfigMap (0.00s)
=== RUN   TestGetPVC
=== RUN   TestGetPVC/TestGetPVC
--- PASS: TestGetPVC (0.00s)
    --- PASS: TestGetPVC/TestGetPVC (0.00s)
=== RUN   TestGetService
=== RUN   TestGetService/TestGetServices
--- PASS: TestGetService (0.00s)
    --- PASS: TestGetService/TestGetServices (0.00s)
=== RUN   TestGetPod
=== RUN   TestGetPod/TestGetPod
--- PASS: TestGetPod (0.00s)
    --- PASS: TestGetPod/TestGetPod (0.00s)
=== RUN   TestGetDeployment
=== RUN   TestGetDeployment/TestGetDeployment
--- PASS: TestGetDeployment (0.00s)
    --- PASS: TestGetDeployment/TestGetDeployment (0.00s)
PASS
ok      github.com/openebs/maya/pkg/client/k8s  0.046s

```

**P.S**
I have now added sub tests. If this looks good to you guys. I can go on and add for maya_test.go

cc / @kmova @AmitKumarDas @prateekpandey14

### CODE REVIEW

**Comment by dungeonmaster18** [comment]
Also I was wondering should we change the go test in makefile to go test -v?

File: pkg/client/k8s/k8s_test.go [hunk 1]
```diff
@@ -35,15 +35,18 @@ func TestGetConfigMap(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		kc := &K8sClient{
-			ConfigMap: test.cm,
-		}
+		t.Run("TestGetConfigMap", func(t *testing.T) {
```
Comment (AmitKumarDas) [inline pkg/client/k8s/k8s_test.go]: Change the type of tests from []struct{..} to map[string]struct{}
The string thing will be the name of the test.

File: pkg/client/k8s/k8s_test.go [hunk 2]
```diff
@@ -58,15 +61,18 @@ func TestGetPVC(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		kc := &K8sClient{
-			PVC: test.pvc,
-		}
+		t.Run("TestGetPVC", func(t *testing.T) {
```
Comment (AmitKumarDas) [inline pkg/client/k8s/k8s_test.go]: Same as above. This logic marks all the tests with same name i.e. TestGetPVC

File: pkg/client/k8s/k8s_test.go [hunk 3]
```diff
@@ -81,15 +87,17 @@ func TestGetService(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		kc := &K8sClient{
-			Service: test.service,
-		}
+		t.Run("TestGetServices", func(t *testing.T) {
```
Comment (AmitKumarDas) [inline pkg/client/k8s/k8s_test.go]: Same as above. Have a look at recent commits to maya to understand more.

File: pkg/client/k8s/k8s_test.go [hunk 4]
```diff
@@ -104,15 +112,17 @@ func TestGetPod(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		kc := &K8sClient{
-			Pod: test.pod,
-		}
+		t.Run("TestGetPod", func(t *testing.T) {
```
Comment (AmitKumarDas) [inline pkg/client/k8s/k8s_test.go]: Same comment applied here.

File: pkg/client/k8s/k8s_test.go [hunk 5]
```diff
@@ -127,14 +137,16 @@ func TestGetDeployment(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		kc := &K8sClient{
-			Deployment: test.deploy,
-		}
+		t.Run("TestGetDeployment", func(t *testing.T) {
```
Comment (AmitKumarDas) [inline pkg/client/k8s/k8s_test.go]: Same comment applied here.

File: pkg/maya/maya_test.go [hunk 6]
```diff
@@ -30,15 +30,17 @@ func TestMayaBytes(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		my := &MayaYaml{
-			Yaml: test.yaml,
-		}
+		t.Run("", func(t *testing.T) {
```
Comment (AmitKumarDas) [inline pkg/maya/maya_test.go]: Same comment applied here.

File: pkg/maya/maya_test.go [hunk 7]
```diff
@@ -58,17 +60,19 @@ ports:
 	}
 
 	for _, test := range tests {
-		mc := NewMayaContainer(test.yaml)
+		t.Run("", func(t *testing.T) {
```
Comment (AmitKumarDas) [inline pkg/maya/maya_test.go]: Same comment applied here.

File: pkg/maya/maya_test.go [hunk 8]
```diff
@@ -89,17 +93,19 @@ ports:
 	}
 
 	for _, test := range tests {
-		mc := NewMayaContainer("")
+		t.Run("", func(t *testing.T) {
```
Comment (AmitKumarDas) [inline pkg/maya/maya_test.go]: Same comment applied here.

File: pkg/maya/maya_test.go [hunk 9]
```diff
@@ -134,17 +140,19 @@ spec:
 	}
 
 	for _, test := range tests {
-		md := NewMayaDeployment(test.yaml)
+		t.Run("", func(t *testing.T) {
```
Comment (AmitKumarDas) [inline pkg/maya/maya_test.go]: Same comment applied here.

File: pkg/maya/maya_test.go [hunk 10]
```diff
@@ -185,20 +193,22 @@ image: openebs/m-apiserver:test
...
-		}
-
-		if test.isErr && len(md.MayaContainer.Container.Name) != 0 {
-			t.Fatalf("Expected: 'nil maya deployment container' Actual: '%v'", md.MayaContainer.Container)
-		}
-
-		if !test.isErr && len(md.Deployment.Spec.Template.Spec.Containers) != 2 {
-			t.Fatalf("Expected: '2 containers in maya deployment' Actual: '%d'", len(md.Deployment.Spec.Template.Spec.Containers))
-		}
+		t.Run("", func(t *testing.T) {
```
Comment (AmitKumarDas) [inline pkg/maya/maya_test.go]: Same comment applied here.

File: pkg/client/k8s/k8s_test.go (hunk 1 above)
Comment (dungeonmaster18) [reply pkg/client/k8s/k8s_test.go]: Okay. I will do that.

Comment (dungeonmaster18) [reply pkg/client/k8s/k8s_test.go]: Actually I was not sure about the test names to put right now. Can you help me with that.

**Comment by dungeonmaster18** [comment]
cc @AmitKumarDas for review.

**Pull request merged**